azimuth/utils/project.py

```diff
@@ -52,6 +53,16 @@ def load_dataset_from_config(azimuth_config: AzimuthConfig) -> DatasetDict:
     return dataset_train_eval
 
 
+def save_config(azimuth_config: AzimuthConfig):
+    """Append config to configs.jsonl to retrieve past configs."""
+    with jsonlines.open(f"{azimuth_config.artifact_path}/configs.jsonl", mode="a") as f:
```

Comment:
The name of the file... I'm afraid plural `configs.jsonl` could be interpreted as multople configs, whereas this is more like the one config, but recording the past versions. What would you think of `config_history.jsonl`?
```suggestion
    with jsonlines.open(f"{azimuth_config.artifact_path}/config_history.jsonl", mode="a") as f:
```

Reply:
Now that it was moved to the parent cache folder, it can also be multiple different configs though. Do you still prefer `config_history`?

Reply:
Right. So, I feel we are going through a transition right now, so all might not be definitive.

I think we need to tie one `.jsonl` per Azimuth instance. If one day we are to start Azimuth without a config (or almost no config), we need a way to identify which `.jsonl` to load. I think that's the `artifact_path`, and that the `jsonl` is in the correct location. Eventually, if one shuts down Azimuth and restarts it (with the same `artifact_path`), I would expect it to load the `.jsonl` and resume with the last config. In that context, if one runs multiple Azimuth instances, I would expect a different `artifact_path` per instance. Considering you could start 2 Azimuth instances with "no config", there is no other way to tell them which `.jsonl` to look at.


Does that make sense?

As I mentioned before, I am questioning the need for the cache "sub-folders". Even if one changes the pipelines completely (for example one adds a pipeline once a model is trained), the dataset stuff that is cached would still be valid.

Reply:
Thanks for further explaining, this makes sense to me!